inbm/dispatcher-agent/dispatcher/schedule/schedules.py

```diff
@@ -9,17 +9,22 @@
 from typing import Optional
 
 @dataclass(init=True)
-class SingleSchedule:
-    """ Represents a SingleSchedule object from the Scheduled Manifest """
+class Schedule:
+    """ Represents a Base class for schedule objects """
     request_id: str
+    id: Optional[int] = field(default=None)    
+
+
+@dataclass(init=True)
+class SingleSchedule(Schedule):
+    """ Represents a SingleSchedule object from the Scheduled Manifest """ 
     start_time: Optional[datetime] = field(default=None)
     end_time: Optional[datetime] = field(default=None)
     manifests: list[str] = field(default_factory=list)
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Add explicit type annotations for manifests list elements for better type checking and clarity.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    from typing import List

    @dataclass
    class SingleSchedule(Schedule):
        manifests: List[str] = field(default_factory=list)

    @dataclass
    class RepeatedSchedule(Schedule):
        manifests: List[str] = field(default_factory=list)
```